sivuyilemene/task-app
# Added pre-commit checks

## Description
This PR introduces pre-commit checks to ensure code quality and consistency before commits are made to the repository. By using Husky, ESLint, and Prettier, I have implemented linting checks that run automatically before a commit is finalized. Additionally, with the integration of Husky and CommitLint, it enforces commit message conventions to follow Semantic Versioning standards.


## Changes Made
- Added Husky pre-commit hooks to run ESLint and Prettier for code linting and formatting.
- Configured ESLint for JavaScript linting.
- Configured Prettier for code formatting.
- Added CommitLint to enforce commit message conventions based on SemVer.


## How to Test
1. Checkout to the branch: git checkout [branch-name]
2. Run `npm i`
3. Make a change to any JavaScript file.
4. Attempt to commit the change with `git commit -m "<type>(optional scope): <description>"`
5. Verify that ESLint and Prettier checks run automatically.
6. Ensure that the commit message follows the Semantic Versioning convention and that any invalid messages are rejected.
